src/herder/TransactionQueue.cpp

```diff
@@ -116,20 +116,44 @@ canReplaceByFee(TransactionFrameBasePtr tx, TransactionFrameBasePtr oldTx,
     return res;
 }
 
+// This method will update iter to point to the tx with seqNum == seq if it is
+// found. It also returns a bool that will be false if it determines seq cannot
+// be added to the current queue, allowing the user of this function to make a
+// decision early if desired.
 static bool
-findBySeq(int64_t seq, TransactionQueue::TimestampedTransactions& transactions,
```

Comment:
I can't write a comment on that line, but `findTx` usage in `ban` might be broken now. Writing some tests for this.

Reply:
After looking at this, I think it's fine.
1. If we try to ban a transaction that fills in a valid gap, that ban should not remove any transactions in the queue.
2. `findTx` will only overwrite `txIter` if there's an exact match on the transaction.